### ISSUE TITLE
Gutenberg lagging and high CPU usage when in paired mode

### ISSUE THREAD

**Issue by asadkn** [Automattic/amp-wp#1679]
When AMP is in paired mode, there's massive lag when using Gutenberg demo post. I have tried this with twentyseventeen theme, and Gutenberg + AMP RC3 plugins active. The problem doesn't occur with default settings but only when paired mode is activated via functions.php:

```php
		// Add support 
		add_theme_support('amp', array(
			'paired' => true
		));
```

Screencast: https://d.pr/v/HLB3BN

**Comment by lkraav** [comment]
Looks very similar to what's happening in https://github.com/WordPress/gutenberg/pull/12312 and the underlying issues there.

**Comment by miina** [comment]
@westonruter Testing locally with a long post on that PR branch: typing is very slow in both Paired and Native mode. With classic mode it seems to be the "normal" experience of Gutenberg. Switched these a few times just in case to make sure it was related to switching the mode and it seemed to be the same every time -- very slow typing in Native / Paired mode.

**Comment by westonruter** [comment]
That's not good! I wonder if it is due to AMP block extensions, or if it is due to the AMP block validation code. Please try disabling one or the other to see if you can narrow down where the problem is.

**Comment by kienstra** [comment]
**Request To Test PR**

Hi @asadkn,
Thanks a lot for bringing up this issue.

Could you please test #1679 to see if it fixes this for you? It worked for me locally.

It looks like you know how this works, but please do `npm run dev` to rebuild the JavaScript files.

**Comment by asadkn** [comment]
Seems to be all good 👍

**Comment by westonruter** [comment]
Merged into 1.0 branch. Will be part of 1.0-RC4. @asadkn Thank you very much for testing.
